src/Controller/BillingController.php

```diff
@@ -94,23 +93,15 @@ public function myPrepaidBalance(): RedirectResponse {
   /**
    * View prepaid balance and account statements, add money to prepaid balance.
    *
-   * @param string $user
-   *   The user id.
-   *
-   * @todo: Use the user object.
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   The Drupal user.
    *
    * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
    *   A render array or a redirect response.
    *
    * @throws \Exception
    */
-  public function prepaidBalancePage(string $user) {
-    $user = User::load($user);
-
-    if (!$user) {
-      throw new NotFoundHttpException();
-    }
-
+  public function prepaidBalancePage(AccountInterface $user) {
```

Comment:
I would have expected that you have to typehint to `UserInterface` to trigger the automatic parameter conversion.

Reply:
Good catch. I changed it to use `UserInterface`.